### ISSUE TITLE
intrusion detection: log error on bootup when not set

### ISSUE THREAD

**Issue by fichtner** [opnsense/core#1212]
(noticed on a 17.1.a test system)

```
configd.py: error generating template OPNsense.IDS : Traceback (most recent call last): File "/usr/local/opnsense/service/modules/template.py", line 318, in generate for filename in self._generate(template_name, create_directory): File "/usr/local/opnsense/service/modules/template.py", line 262, in _generate raise Exception("%s %s %s" % (module_name, template_filename, render_exception)) Exception: OPNsense.IDS OPNsense/IDS/suricata.yaml 'collections.OrderedDict object' has no attribute 'IDS'
```

**Comment by AdSchellevis** [comment]
@fichtner did you run a normal upgrade? Our model migration should generate the structure

**Comment by fichtner** [comment]
@AdSchellevis this was on a major upgrade, maybe there is some issue with php not running correctly, I'll have to see

**Comment by fichtner** [comment]
I can reproduce this during testing of 16.7.6:

```
configd.py: error generating template OPNsense.IDS : Traceback (most recent call last): File "/usr/local/opnsense/service/modules/template.py", line 318, in generate for filename in self._generate(template_name, create_directory): File "/usr/local/opnsense/service/modules/template.py", line 262, in _generate raise Exception("%s %s %s" % (module_name, template_filename, render_exception)) Exception: OPNsense.IDS OPNsense/IDS/suricata.yaml 'OPNsense' is undefined
```

**Comment by AdSchellevis** [comment]
@fichtner I think we miss /usr/local/opnsense/mvc/script/run_migrations.php in +POST_INSTALL, which automatically makes sure the structure exists, so you don't have to check in the template...

**Comment by fichtner** [comment]
We don't have the new migration code in 16.7.x and I could also reproduce this with -devel on fresh installs. It seems the migration is not run if "OPNsense" is not in the config?

**Comment by AdSchellevis** [comment]
it seems that I forgot to add the migration to +POST_INSTALL, which automatically checks the config structure.  for 16.7.x we can't fix this without checking the template, dev should be able to use the structure without validation, as long as the migration code is run after upgrade.

**Comment by fichtner** [comment]
It's not in +POST_INSTALL anymore: https://github.com/opnsense/core/commit/c78c9ced21924806b6578e8c0e1bf782c2f35ffe

**Comment by AdSchellevis** [comment]
@fichtner I missed that one, it should be in there...

**Comment by AdSchellevis** [comment]
there's no problem in also running this on config migration, but upgrades can change models and therefore should always trigger migrations.

**Comment by fichtner** [comment]
it's called indirectly by rc.configure_firmware -> convert_config()

**Comment by AdSchellevis** [comment]
ok, that's odd..  can you try running /usr/local/opnsense/mvc/script/run_migrations.php on your box and check for the existence of OPNsense.IDS.general.syslog ?
https://github.com/opnsense/core/blob/master/src/opnsense/mvc/app/models/OPNsense/IDS/IDS.xml#L150 should force the value to 0 when new.

**Comment by fichtner** [comment]
proxy migration works, I upgraded to the test package for 16.7.6, where the problem exists, then installed opnsense-devel where the latest migration code is:

<img width="549" alt="screen shot 2016-10-10 at 8 26 47 pm" src="https://cloud.githubusercontent.com/assets/1915288/19246730/1248d2c4-8f28-11e6-9a05-a1e6ed7addc5.png">

I guess since IDS does not have an initial revision it's not going to get created.

In any way, I fixed up the packages for 16.7.6 with the old helper.exists thingy. :)

**Comment by AdSchellevis** [comment]
ok, my mistake I forgot to add a version to IDS :) when the migration code is in our production version, we should remove a lot of the helper.exists  things for readability, but no rush there. thanks for fixing.
